code/game/gamemodes/game_mode.dm

```diff
@@ -323,18 +323,18 @@
 	for(var/datum/mind/mind in candidates)
 		p_ckey = ckey(mind.key)
 		p_rep = SSpersistence.antag_rep[p_ckey]
-		p_rep = p_rep == null ? 0 : p_rep
 
-		if(current <= antag_select)
-			var/subtract = min(p_rep + DEFAULT_ANTAG_TICKETS, MAX_TICKETS_PER_ROLL) - DEFAULT_ANTAG_TICKETS
-			SSpersistence.antag_rep_change[p_ckey] = -subtract
+		var/previous = current
+		var/spend = min(p_rep + DEFAULT_ANTAG_TICKETS, MAX_TICKETS_PER_ROLL)
+		current += spend
+
+		if(antag_select >= previous && antag_select <= (current-1))

```

Comment:
This looks good, yep.

Though it might be more readable to have this as
`if(antag_select <= current)`
dropping `previous` and initializing `current` to 0 instead, since that'd hit the same range of values.

Reply:
hello o/

Reply:
hello \o